### PR TITLE
master

### PR DESCRIPTION
**Pull request by vasap87**
Added /vufs part to swagger description file.

### CODE REVIEW

File: doc/swagger-api-description/api.yaml [hunk 1]
```diff
@@ -445,36 +589,153 @@ definitions:
...
+      uuid:
+        type: string
+        description: Allocation model UUID.
+      modelType:
+        description: Types of allocation model.
+        enum:
+          - UNION
+          - MORROR
+          - HIGH_AVAILABILITY
+      propertySet:
```
Comment (ancord-m) [inline doc/swagger-api-description/api.yaml]: Why did you inculde file's properties in allocation model?

File: doc/swagger-api-description/api.yaml [hunk 2]
```diff
@@ -445,36 +589,153 @@ definitions:
...
+          $ref: "#/definitions/Property"
+
+  Statistics:
+    type: object
+    description: Contain statistic information about cloud.
+    properties:
+      uuid:
+        type: string
+        description: Statistic UUID.
+      cloudSet:
```
Comment (ancord-m) [inline doc/swagger-api-description/api.yaml]: Why do you call it cloudSet?

File: doc/swagger-api-description/api.yaml [hunk 3]
```diff
@@ -445,36 +589,153 @@ definitions:
...
+      downloadFiles:
+        description: Count of downloaded files.
+        type: integer
+        format: int32
+      downloadTimes:
+        description: Time what files was downloaded.
+        type: integer
+        format: int64
+
+  HelthStatistis:
```
Comment (ancord-m) [inline doc/swagger-api-description/api.yaml]: As I have written above: very abstract object. Please explain what it is going to be used for. 
_In a wisper_ : please correct spelling (he _a_ lth, spa _c_ e). Sorry for this.

File: doc/swagger-api-description/api.yaml [hunk 4]
```diff
@@ -445,36 +589,153 @@ definitions:
...
+        type: array
+        items:
+          $ref: "#/definitions/CloudSpase"
+
+  CloudSpase:
+    type: object
+    description: Contain information about cloud availability space.
+    properties:
+      uuid:
+        type: string
```
Comment (ancord-m) [inline doc/swagger-api-description/api.yaml]: Do we need a Uuid here? a _cloud_ under _ref_ param already contains it's own Uuid so we know which cloud we are talking about and it's available space.

File: core/core.iml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kosmoflyko) [inline core/core.iml]: Please rebase you fork. Common project contain now .gitignore files with rule ignone *.iml

File: doc/swagger-api-description/api.yaml [hunk 6]
```diff
@@ -245,6 +245,195 @@ paths:
...
+        '404':
+          description: Bad user UUID.
+          schema:
+            $ref: '#/definitions/ErrorModel'
+        default:
+          description: Unexpected error.
+          schema:
+            $ref: '#/definitions/ErrorModel'
+    post:
+    #QUESTION: Can we create VUFS snapshot with changes in local file storage on client side? Or we send a 'signal' about changes and server check what exactly has changed. And as response send VUFS diff object or full VUFS object?
```
Comment (kosmoflyko) [inline doc/swagger-api-description/api.yaml]: Diff can make client and server.
Case 1 - If when client receive event from local file storage monitoring services, VUFS generate diff. But this diff contain files not upload on clouds. Client send request with the diff to server and get response from server with the diff contains advanced allocate information  (where server write allocate space for each file part).
Case 2 - If server receive event from cloud monitoring about change files, VUFS on server make diff contains changes and send him on clients as push notification.

File: doc/swagger-api-description/api.yaml [hunk 7]
```diff
@@ -371,6 +560,183 @@ definitions:
...
+        description: Virtual file UUID.
+      file-properties:
+        type: array
+        items:
+          $ref: "#/definitions/Property"
+      file-parts:
+        type: array
+        items:
+          $ref: "#/definitions/PartLocation"
+      VirtualChildSet:
```
Comment (kosmoflyko) [inline doc/swagger-api-description/api.yaml]: Change property name to "ChildSet"

File: doc/swagger-api-description/api.yaml [hunk 8]
```diff
@@ -371,6 +560,183 @@ definitions:
...
+        type: string
+        description: CloudSpase UUID.
+      cloud:
+        $ref: "#/definitions/Cloud"
+      availableSpace:
+        description: Cloud available spase in bytes.
+        type: integer
+        format: int64
+
+  AllocateModel:
```
Comment (kosmoflyko) [inline doc/swagger-api-description/api.yaml]: Where  we use this object?

File: doc/swagger-api-description/api.yaml (hunk 3 above)
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: Thanks for this.

File: doc/swagger-api-description/api.yaml (hunk 2 above)
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: I change it to _cloud_, thanks.

File: doc/swagger-api-description/api.yaml (hunk 1 above)
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: _Property_ is reused object. It's key-value object.

File: doc/swagger-api-description/api.yaml [hunk 9]
```diff
@@ -371,6 +560,183 @@ definitions:
...
+    type: object
+    properties:
+      uuid:
+        type: string
+        description: Allocation model UUID.
+      modelType:
+        description: Types of allocation model.
+        enum:
+          - UNION
+          - MORROR
```
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: I checked it. Thanks.

File: doc/swagger-api-description/api.yaml (hunk 7 above)
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: Done

File: doc/swagger-api-description/api.yaml [hunk 10]
```diff
@@ -371,6 +560,183 @@ definitions:
...
+        format: date-time
+      owner:
+        description: Owner of this snapshot.
+        $ref: '#/definitions/User'
+      virtualFileSet:
+        type: array
+        items:
+          $ref: "#/definitions/VirtualFile"
+
+  VirtualFile:
```
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: I agree with you. Changed name from virtual to vufs

File: doc/swagger-api-description/api.yaml [hunk 11]
```diff
@@ -292,7 +292,147 @@ paths:
...
+        default:
+          description: Unexpected error.
+          schema:
+            $ref: '#/definitions/ErrorModel'
+
+  /vufs/space/{userUuid}:
+    get:
+      description: Allows to get availability space on each cloud.
+      operationId: getAvailableSpace
+      parameters:
```
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: Ok. Done

File: doc/swagger-api-description/api.yaml [hunk 12]
```diff
@@ -292,7 +292,147 @@ paths:
...
+        '404':
+          description: Bad cloud UUID.
+          schema:
+            $ref: '#/definitions/ErrorModel'
+        default:
+          description: Unexpected error.
+          schema:
+            $ref: '#/definitions/ErrorModel'
+
+  /vufs/stat/helth/{userUuid}:
```
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: Yes, server do it automatically, but how clients take it without API?

File: doc/swagger-api-description/api.yaml [hunk 13]
```diff
@@ -292,7 +292,147 @@ paths:
...
+        default:
+          description: Unexpected error.
+          schema:
+            $ref: '#/definitions/ErrorModel'
+
+  /vufs/stat/{cloudUuid}:
+    get:
+      description: Allows to get statistic object about selected cloud.
+      operationId: getCloudStatistic
+      parameters:
```
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: I change paramert name to userUuid

File: doc/swagger-api-description/api.yaml [hunk 14]
```diff
@@ -371,6 +578,202 @@ definitions:
...
+      uuid:
+        type: string
+        description: CloudSpaseSet UUID.
+        format: uuid
+      cloudSpaceSet:
+        type: array
+        items:
+          $ref: "#/definitions/CloudSpase"
+
+  CloudSpase:
```
Comment (kosmoflyko) [inline doc/swagger-api-description/api.yaml]: Remove this object,  use Cloud object and extend him if need

File: doc/swagger-api-description/api.yaml [hunk 15]
```diff
@@ -371,6 +578,202 @@ definitions:
...
+    type: object
+    description:  Contain key-value information about cloud and its helth status.
+    properties:
+      cloud:
+        $ref: "#/definitions/Cloud"
+      isAvailable:
+        description: Is cloud available?
+        type: boolean
+
+  CloudSpaseSet:
```
Comment (kosmoflyko) [inline doc/swagger-api-description/api.yaml]: Remove this object,  use CloudSet object and extend him if need

File: doc/swagger-api-description/api.yaml (hunk 3 above)
Comment (kosmoflyko) [inline doc/swagger-api-description/api.yaml]: Remove this object. Use Statistic and StatisticSet object with keys-value:
[{uuid: 'clouds's uuid'}, {available: true}]

File: doc/swagger-api-description/api.yaml [hunk 16]
```diff
@@ -371,6 +578,202 @@ definitions:
...
+      uuid:
+        type: string
+        description: HelthStatistis UUID.
+        format: uuid
+      helthCloudSet:
+        type: array
+        items:
+          $ref: "#/definitions/HelthCloud"
+
+  HelthCloud:
```
Comment (kosmoflyko) [inline doc/swagger-api-description/api.yaml]: Remove this object. Use Statistic and StatisticSet object with keys-value:
[{uuid: 'clouds's uuid'}, {available: true}]

File: doc/swagger-api-description/api.yaml [hunk 17]
```diff
@@ -371,6 +578,202 @@ definitions:
...
+        description: InitialAllocateModel UUID.
+        format: uuid
+      file:
+        $ref: "#/definitions/VufsFile"
+      allocateModel:
+        $ref: "#/definitions/AllocationModel"
+
+  StatisticSet:
+    type: object
+    description: Contain set of statistic information about cloud.
```
Comment (kosmoflyko) [inline doc/swagger-api-description/api.yaml]: Fix description: "Contain set of statistic information about API object".

File: doc/swagger-api-description/api.yaml [hunk 18]
```diff
@@ -371,6 +578,202 @@ definitions:
...
+    properties:
+      statisticSet:
+        type: array
+        items:
+          $ref: "#/definitions/Statistic"
+
+  Statistic:
+    type: object
+    description: Contain statistic information about cloud.
```
Comment (kosmoflyko) [inline doc/swagger-api-description/api.yaml]: Fix description: "Contain statistic information about API object".

File: doc/swagger-api-description/api.yaml [hunk 19]
```diff
@@ -245,6 +245,213 @@ paths:
...
+        '404':
+          description: Bad user UUID.
+          schema:
+            $ref: '#/definitions/ErrorModel'
+        default:
+          description: Unexpected error.
+          schema:
+            $ref: '#/definitions/ErrorModel'
+
+  /vufs/allocation:
```
Comment (kosmoflyko) [inline doc/swagger-api-description/api.yaml]: Add template .../{userUuid} in path

File: doc/swagger-api-description/api.yaml [hunk 20]
```diff
@@ -371,6 +578,202 @@ definitions:
...
+          $ref: "#/definitions/PartLocation"
+      childSet:
+        type: array
+        items:
+          $ref: "#/definitions/VufsFile"
+      allocateModel:
+        $ref: "#/definitions/AllocationModel"
+
+  Property:
+    description: Describes all file property.
```
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: Done

File: doc/swagger-api-description/api.yaml [hunk 21]
```diff
@@ -371,6 +578,202 @@ definitions:
...
+    type: object
+    description: Contain statistic information about cloud.
+    properties:
+      uuid:
+        type: string
+        description: Statistic UUID.
+        format: uuid
+      cloud:
+        $ref: "#/definitions/Cloud"
+      uploadFiles:
```
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: Done.

File: doc/swagger-api-description/api.yaml (hunk 18 above)
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: Done.

File: doc/swagger-api-description/api.yaml (hunk 17 above)
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: Done.

File: doc/swagger-api-description/api.yaml (hunk 15 above)
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: Done.

File: doc/swagger-api-description/api.yaml (hunk 16 above)
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: Removed health

File: doc/swagger-api-description/api.yaml [hunk 22]
```diff
@@ -445,36 +589,153 @@ definitions:
...
       file-properties:
         type: array
         items:
-          $ref: "#/definitions/FileProperty"
+          $ref: "#/definitions/Property"
       file-parts:
         type: array
         items:
           $ref: "#/definitions/PartLocation"
+      VirtualChildSet:
```
Comment (vasap87) [inline doc/swagger-api-description/api.yaml]: Parent or Child are similar to creating tree structure.

File: doc/swagger-api-description/api.yaml [hunk 23]
```diff
@@ -329,19 +535,28 @@ definitions:
...
     required:
       - cloudType
       - cloudUrl
       - cloudAccessToken
+      - availableSpace
 
   CloudSet:
     description: A representation of a cloud list that user has.
     type: array
+    allOf:
```
Comment (kosmoflyko) [inline doc/swagger-api-description/api.yaml]: Remove "allOf". Collection not contain UUID.

File: doc/swagger-api-description/api.yaml [hunk 24]
```diff
@@ -329,19 +535,28 @@ definitions:
...
+      - $ref: '#/definitions/AriaddnaObj'
     items:
       $ref: '#/definitions/Cloud'
 
   CloudSetPages:
     description: This page-object shall be used for visual representation of a list (which is inside a CloudSet) of user's external cloud accounts.
+    allOf:
```
Comment (kosmoflyko) [inline doc/swagger-api-description/api.yaml]: Remove "allOf". Collection not contain UUID.

File: doc/swagger-api-description/api.yaml [hunk 25]
```diff
@@ -366,14 +583,158 @@ definitions:
...
+      allocationModel:
+        $ref: '#/definitions/AllocationModel'
+    required:
+      - file
+      - allocationModel
+
+  StatisticSet:
+    type: object
+    description: Contain set of statistic information about API object.
+    allOf:
```
Comment (kosmoflyko) [inline doc/swagger-api-description/api.yaml]: Remove "allOf". Collection not contain UUID.

File: doc/swagger-api-description/api.yaml [hunk 26]
```diff
@@ -366,14 +583,158 @@ definitions:
...
+        type: array
+        items:
+          $ref: '#/definitions/Property'
+    required:
+      - cloud
+
   ErrorModel:
     description: Information about error.
     type: object
+    allOf:
```
Comment (kosmoflyko) [inline doc/swagger-api-description/api.yaml]: Remove "allOf". Error not contain UUID.

**Pull request merged**